tutorials/Window_API_new/README.md

```diff
@@ -0,0 +1,492 @@
+    val windows = mutableStateListOf<MyWindowState>()
+
+    init {
+        windows += MyWindowState("Initial window")
+    }
+
+    fun openNewWindow() {
+        windows += MyWindowState("Window ${windows.size}")
+    }
+
+    fun exit() {
+        windows.clear()
+    }
+
+    private fun MyWindowState(
+        title: String
+    ) = MyWindowState(
+        title,
+        openNewWindow = ::openNewWindow,
+        exit = ::exit,
+        windows::remove
+    )
+}
+
+private class MyWindowState(
+    val title: String,
+    val openNewWindow: () -> Unit,
+    val exit: () -> Unit,
+    private val close: (MyWindowState) -> Unit
+) {
+    fun close() = close(this)
+}
+```
+(don't forget to define a custom `onCloseRequest` to remove a window from the list)
+![](multiple_windows.gif)
+
+## Changing state (maximized, minimized, fullscreen, size, position) of the window.
+
+Some state of the window is hoisted into a separate class `WindowState`. You can change its properties in callbacks or observe it in Composable's:
+
+```kotlin
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.material.Checkbox
+import androidx.compose.material.Text
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.ExperimentalComposeUiApi
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.window.Window
+import androidx.compose.ui.window.application
+import androidx.compose.ui.window.rememberWindowState
+
+@OptIn(ExperimentalComposeUiApi::class)
+fun main() = application {
+    val state = rememberWindowState(isMaximized = true)
+
+    Window(state) {
+        Column {
+            Row(verticalAlignment = Alignment.CenterVertically) {
+                Checkbox(state.isFullscreen, { state.isFullscreen = !state.isFullscreen })
+                Text("isFullscreen")
+            }
+
+            Row(verticalAlignment = Alignment.CenterVertically) {
+                Checkbox(state.isMaximized, { state.isMaximized = !state.isMaximized })
+                Text("isMaximized")
+            }
+
+            Row(verticalAlignment = Alignment.CenterVertically) {
+                Checkbox(state.isMinimized, { state.isMinimized = !state.isMinimized })
+                Text("isMinimized")
+            }
+
+            Text(
+                "Position ${state.position}",
+                Modifier.clickable {
+                    state.position = state.position.copy(x = state.position.x + 10.dp)
+                }
+            )
+
+            Text(
+                "Size ${state.size}",
+                Modifier.clickable {
+                    state.size = state.size.copy(width = state.size.width + 10.dp)
+                }
+            )
+        }
+    }
+}
+```
+![](state.gif)
+
+## Handle window-level shortcuts
+(it is not a final variant, we will investigate how can we provide a simple API for that):
```

Comment:
Shall we remove the comment?

Reply:
Changed this comment and moved it below the snippet. I think it is better to keep it to avoid confusion about verbosity.